internal/access/identity.go

```diff
@@ -86,13 +86,47 @@ func DeleteIdentity(c *gin.Context, id uid.ID) error {
 	return data.DeleteIdentity(db, id)
 }
 
-func ListIdentities(c *gin.Context, name string) ([]models.Identity, error) {
+func ListIdentities(c *gin.Context, name string, includeUnlinked bool) ([]models.Identity, error) {
 	db, err := RequireInfraRole(c, models.InfraAdminRole, models.InfraViewRole, models.InfraConnectorRole)
 	if err != nil {
 		return nil, err
 	}
 
-	return data.ListIdentities(db, data.ByOptionalName(name))
+	identities, err := data.ListIdentities(db, data.ByOptionalName(name))
+	if err != nil {
+		return nil, err
+	}
+
+	if includeUnlinked {
+		return identities, nil
+	}
+
+	// filter out identities that do not have a linked provider
+
+	providerIdentities, err := data.ListProviderUsers(db)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
this is potentially too expensive. you should instead query for ProviderUsers that are in the list of returned users and return only those.